kube/services/revproxy/helpers.js

```diff
@@ -253,3 +253,29 @@ function checkBlackList(req,res) {
   }
   return "ok"; // + "-" + ipAddrStr + "-" + blackListStr;
 }
+
+
+/**
+ * Handle the js_content callout from /workspace-authorize.
+ * Basically - redirect to a subdomain /wts/authorize endpoint
+ * based on the state=SUBDOMAIN-... query parameter with
+ * some guards to stop attacks.
+ * 
+ * @param {*} req 
+ * @param {*} res 
+ */
+function gen3_workspace_authorize_handler(req) {
+  var subdomain = '';
+  var query = req.variables["args"] || "";
+  var matchGroups = null;
+
+  if (matchGroups = query.match(/(^state=|&state=)(\w+)-/)) {
```

Comment:
is there a technical reason for limiting to w+?

Reply:
I didn't want to leave open the possibility of a XSS scripting attack where an attacker sends a link to a victim something like:
    https://workspace.gen3.org/workspace-authorize/?state=https://evil.com/-....